Over21/ViewController.swift

```diff
@@ -128,6 +139,70 @@ extension ViewController: CaptureHelperDevicePresenceDelegate {
         print("scanner arrived")
         ageIndicatorView.updateScannerConnection(isConnected: true)
         
+        getDataSource(with: device) { [weak self] (result) in
+            guard let strongSelf = self else { return }
+            
+            if result == SKTResult.E_NOTSUPPORTED {
+                DispatchQueue.main.async {
+                    strongSelf.notificationsView.setMessage(to: "The connected device cannot scan barcodes.")
+                    strongSelf.notificationsView.animate()
+                }
+            } else if result == SKTResult.E_NOERROR {
+                strongSelf.setScanButtonNotifications(with: device)
+            }
+        }
+    
+    }
+    
+    func didNotifyRemovalForDevice(_ device: CaptureHelperDevice, withResult result: SKTResult) {
+        print("scanner removed")
+        ageIndicatorView.updateScannerConnection(isConnected: false)
+    }
+    
+    private func getDataSource(with device: CaptureHelperDevice, completionHandler: @escaping (_ result: SKTResult) -> ()) {
+        
+        device.getDataSourceInfoFromId(SKTCaptureDataSourceID.symbologyPdf417) { (result, captureDataSource) in
+    
+            if result == SKTResult.E_NOTSUPPORTED {
```

Comment:
I think this code could be simplified. In all cases we call `completionHandler(result)`, and in the case where the PDF417 is not enabled then we return the `completionHandler(result)` upon the completion of enabling the PDF417 data source.

Reply:
Will there ever be a case where PDF417 is supported but not currently enabled? `(result == SKTResult.E_NOERROR) and captureDataSource.status == .disabled)`
That's why that `case` is the only one that waits to enable before returning the `completionHandler`